src/components/EggheadPlayer/use-egghead-player.ts

```diff
@@ -212,44 +213,69 @@ const onEnded = (lesson: {
   if (lesson.lesson_view_url) {
     onComplete(lesson)
   }
-
-  track('finished lesson', {
-    lesson: lesson.slug,
-    tags: lesson.tags.map((tag: {slug: any}) => tag.slug),
-  })
 }
 
 function onError() {}
 
-function onVideoQualityChanged() {}
-
-function onSubtitleChange() {}
-
-const getPlayerPrefs = () => cookies.get('egghead-player-prefs')
+const PLAY_PREFS_KEY = 'egghead-player-prefs'
+
+const defaultPlayerPreferences = {
+  volumeRate: 80,
+  playbackRate: 1,
+  autoplay: false,
+  videoQuality: {
+    bitrate: null,
+    height: null,
+    id: 'auto',
+    width: null,
+  },
+  subtitle: {
+    id: null,
+    kind: null,
+    label: 'off',
+    lang: null,
+  },
+  muted: false,
+}
 
-function onPlaybackRateChange(playbackRate: number) {
-  cookies.set('egghead-player-prefs', {...getPlayerPrefs(), playbackRate})
+export const getPlayerPrefs = () => {
+  return (
+    cookies.get(PLAY_PREFS_KEY) ||
+    cookies.set(PLAY_PREFS_KEY, defaultPlayerPreferences)
+  )
 }
 
-function onVolumeChange(volumeRate: number) {
-  cookies.set('egghead-player-prefs', {...getPlayerPrefs(), volumeRate})
+export const savePlayerPrefs = (options: any) => {
+  console.log(`storing preference`, options)
+  return cookies.set(PLAY_PREFS_KEY, {
+    ...defaultPlayerPreferences,
+    ...getPlayerPrefs(),
+    ...options,
```

Comment:
we pass in an `options` object like `{muted: true}` and it saves the whole prefs